mlflow/metrics/utils/make_genai_metric.py

```diff
@@ -160,31 +160,31 @@ def eval_fn(eval_df: Union["pd.DataFrame", "pyspark.sql.DataFrame"]) -> MetricVa
 
         # TODO: Save the metric definition in a yaml file for model monitoring
 
-        messages = []
+        payload = []
         for indx, (input, output) in enumerate(zip(inputs, outputs)):
             variable_string = _format_variable_string(variables, eval_df, indx)
-            messages.append(
-                [
-                    evaluation_context["eval_prompt"].format(
+            payload.append(
+                {
+                    "prompt": evaluation_context["eval_prompt"].format(
                         input=input, output=output, variables=variable_string
                     ),
-                ],
+                    **eval_parameters,
+                },
             )
 
-        # TODO: load the model for openai and gateway URI
+        eval_result = None
         if isinstance(eval_model, str):
-            eval_model = _load_model_or_server(eval_model, _EnvManager.LOCAL)
-        elif isinstance(eval_model, PyFuncModel):
-            pass
+            eval_model = model_utils.score_model_on_payload(eval_model, _EnvManager.LOCAL)
+            # TODO: Add batch processing for messages here
+            eval_result = eval_model(payload)
```

Comment:
is this usage correct? i think we need to do something like `model_utils.score_model_on_payload(eval_model, payload)`